src/dex/cables/cables.ts

```diff
@@ -0,0 +1,1006 @@
+  }
+
+  async getCachedPairs(): Promise<any> {
+    const cachedPairs = await this.dexHelper.cache.get(
+      this.dexKey,
+      this.network,
+      this.rateFetcher.pairsCacheKey,
+    );
+
+    return cachedPairs ? JSON.parse(cachedPairs) : {};
+  }
+
+  async getCachedPrices(): Promise<any> {
+    const cachedPrices = await this.dexHelper.cache.get(
+      this.dexKey,
+      this.network,
+      this.rateFetcher.pricesCacheKey,
+    );
+
+    return cachedPrices ? JSON.parse(cachedPrices) : {};
+  }
+
+  async getCachedTokensAddr(): Promise<any> {
+    const tokens = await this.getCachedTokens();
+    const tokensAddr: Record<string, Address> = {};
+    for (const key of Object.keys(tokens)) {
+      tokensAddr[tokens[key].symbol.toLowerCase()] = tokens[key].address;
+    }
+    return tokensAddr;
+  }
+
+  getPairString(baseToken: Token, quoteToken: Token): string {
+    return `${baseToken.symbol}/${quoteToken.symbol}`.toLowerCase();
+  }
+
+  // Function to find a key by address
+  private findKeyByAddress = (
+    jsonData: Record<string, { address: string }>,
+    targetAddress: string,
+  ): string | undefined => {
+    const entries = Object.entries(jsonData);
+    const foundEntry = entries.find(
+      ([_, value]) =>
+        value.address.toLowerCase() === targetAddress.toLowerCase(),
+    );
+    return foundEntry ? foundEntry[0] : undefined;
+  };
+
+  async getPairData(srcToken: Token, destToken: Token): Promise<any> {
+    const normalizedSrcToken = this.normalizeToken(srcToken);
+    const normalizedDestToken = this.normalizeToken(destToken);
+
+    if (normalizedSrcToken.address === normalizedDestToken.address) {
+      return null;
+    }
+
+    const cachedTokens = await this.getCachedTokens();
+
+    normalizedSrcToken.symbol = this.findKeyByAddress(
+      cachedTokens,
+      normalizedSrcToken.address,
+    );
+    normalizedDestToken.symbol = this.findKeyByAddress(
+      cachedTokens,
+      normalizedDestToken.address,
+    );
+
+    const cachedPairs = await this.getCachedPairs();
+
+    const potentialPairs = [
+      {
+        base: normalizedSrcToken.symbol,
+        quote: normalizedDestToken.symbol,
+        identifier: this.getPairString(normalizedSrcToken, normalizedDestToken),
+        isSrcBase: true,
+      },
+      {
+        base: normalizedDestToken.symbol,
+        quote: normalizedSrcToken.symbol,
+        identifier: this.getPairString(normalizedDestToken, normalizedSrcToken),
+        isSrcBase: false,
+      },
+    ];
+
+    for (const pair of potentialPairs) {
+      if (pair.identifier in cachedPairs) {
+        const pairData = cachedPairs[pair.identifier];
+        pairData.isSrcBase = pair.isSrcBase;
+        return pairData;
+      }
+    }
+    return null;
+  }
+
+  async isBlacklisted(txOrigin: Address): Promise<boolean> {
+    const cachedBlacklist = await this.dexHelper.cache.get(
+      this.dexKey,
+      this.network,
+      CABLES_BLACKLIST_CACHE_KEY,
```

Comment:
Could it be replaced to `this.rateFetcher.blacklistCacheKey` so we don't need separate `CABLES_BLACKLIST_CACHE_KEY` variable?

Reply:
fixed